pkg/test/integration/common/framework.go

```diff
@@ -1242,10 +1250,10 @@ func (c *IntegrationTestFramework) Cleanup() {
 				Machines(controlClusterNamespace).
 				Watch(ctx, metav1.ListOptions{TimeoutSeconds: &timeout}) //ResourceVersion: machineObj.ResourceVersion
 			for event := range watchMachines.ResultChan() {
-				c.ControlCluster.McmClient.
+				Expect(c.ControlCluster.McmClient.
```

Comment:
can you ignore `NotFound` errors here and similar other cases . 
Example https://github.com/gardener/gardener/blob/d36d8944d19040f4f6ddd2043dbe93c401291a13/test/integration/gardenlet/bastion/bastion_test.go#L265

Reply:
Using `.To(Or(Succeed(), BeNotFoundError()))` the same way here will require a `gardener/gardener` import which is a huge change. Let me refactor that in a way that we can live here without for now.